torchvision/datasets/flowers102.py

```diff
@@ -0,0 +1,116 @@
+from pathlib import Path
+from typing import Any, Tuple, Callable, Optional
+
+import numpy as np
+import PIL.Image
+
+from .utils import verify_str_arg, download_and_extract_archive, download_url
+from .vision import VisionDataset
+
+
+class Flowers102(VisionDataset):
+    """`Oxford 102 Flower <https://www.robots.ox.ac.uk/~vgg/data/flowers/102/>`_ Dataset.
+
+    .. warning::
+
+        This class needs `scipy <https://docs.scipy.org/doc/>`_ to load target files from `.mat` format.
+
+    Oxford 102 Flower is an image classification dataset consisting of 102 flower categories. The
+    flowers chosen to be flower commonly occurring in the United Kingdom. Each class consists of
+    between 40 and 258 images.
+
+    The images have large scale, pose and light variations. In addition, there are categories that
+    have large variations within the category and several very similar categories.
+
+    Args:
+        root (string): Root directory of the dataset.
+        split (string, optional): The dataset split, supports ``"train"`` (default), ``"val"``, or ``"test"``.
+        transform (callable, optional): A function/transform that takes in an PIL image and returns a
+            transformed version. E.g, ``transforms.RandomCrop``.
+        target_transform (callable, optional): A function/transform that takes in the target and transforms it.
+    """
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        download: bool = True,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+    ) -> None:
+        super().__init__(root, transform=transform, target_transform=target_transform)
+        self._split = verify_str_arg(split, "split", ("train", "valid", "test"))
+        self._base_folder = Path(self.root) / "flowers-102"
+        self._meta_folder = self._base_folder / "labels"
+        self._images_folder = self._base_folder / "jpg"
+
+        if download:
+            self._download()
+
+        if not self._check_exists():
+            raise RuntimeError("Dataset not found. You can use download=True to download it")
+
+        self._labels = []
+        self._image_files = []
+
+        from scipy.io import loadmat
+
+        # Read the label ids
+        label_mat = loadmat(self._meta_folder / "imagelabels.mat")
+        labels = label_mat["labels"][0]
+
+        self.classes = np.unique(labels).tolist()
+        self.class_to_idx = dict(zip(self.classes, range(len(self.classes))))
```

Comment:
`self.classes` is traditionally reserved for human-readable names. What you define here would be more fitting for `self.labels`.

If I'm not mistaken, there is no way to get the human readable classes from the original data. So we could either use some external mapping such as [this](https://gist.github.com/JosephKJ/94c7728ed1a8e0cd87fe6a029769cde1) or [this](https://github.com/bdevnani3/oxfordflowers102-label-name-mapping/blob/main/mapping.json), but then we would also need to make sure that this is correct and from a quick glance, both variants are not the same. The other option is to simply drop the human-readable classes outright, since the authors didn't provide them either. I would go for the latter.

Reply:
Hi @pmeier , I rename `self.classes` to `self.labels` here.

Reply:
I think you misunderstood me. As is, `self.labels` is a list of all available labels. That is not useful. Either you provide human readable strings for the classes or simply remove `self.labels` and `self.label_to_index`. Since there is no official mapping between labels and class names, I would go for the latter and remove the attributes.